third_party/nvidia/backend/compiler.py

```diff
@@ -53,6 +53,12 @@ def ptx_get_version(cuda_version) -> int:
     raise RuntimeError("Triton only support CUDA 10.0 or higher")
 
 
+@functools.lru_cache(None)
+def file_hash(path):
+    with open(path, "rb") as f:
+        return hashlib.md5(f.read()).hexdigest()
```

Comment:
I know we use md5 elsewhere, but it's a really, really bad hash.  Could we use sha256?

Reply:
Sure thing!  I'll use sha256 here, and then would it be ok if I sent a follow up to convert other uses of md5 to sha256?

Reply:
Absolutely, and thank you!  (I'm moderately less worried about the other md5 because it'd be harder to force a collision, since it's based on the hash of the template below.  But for the sake of #nevermd5 we should still do it :D.)